models/repo/pushmirror.go

```diff
@@ -63,34 +83,41 @@ func UpdatePushMirror(m *PushMirror) error {
 	return err
 }
 
-// DeletePushMirrorByID deletes a push-mirrors by ID
-func DeletePushMirrorByID(ID int64) error {
-	_, err := db.GetEngine(db.DefaultContext).ID(ID).Delete(&PushMirror{})
-	return err
-}
-
 // DeletePushMirrorsByRepoID deletes all push-mirrors by repoID
 func DeletePushMirrorsByRepoID(repoID int64) error {
 	_, err := db.GetEngine(db.DefaultContext).Delete(&PushMirror{RepoID: repoID})
 	return err
 }
 
-// GetPushMirrorByID returns push-mirror information.
-func GetPushMirrorByID(ID int64) (*PushMirror, error) {
-	m := &PushMirror{}
-	has, err := db.GetEngine(db.DefaultContext).ID(ID).Get(m)
+func DeletePushMirrors(opts PushMirrorOptions) error {
+	if opts.RepoID > 0 {
+		_, err := db.GetEngine(db.DefaultContext).Where(opts.toConds()).Delete(&PushMirror{})
```

Comment:
I'm looking at the references of this function, I think we can add `ctx` here.
```suggestion
func DeletePushMirrors(ctx context.Context, opts PushMirrorOptions) error {
	if opts.RepoID > 0 {
		_, err := db.GetEngine(ctx).Where(opts.toConds()).Delete(&PushMirror{})
```
Please also update all references to pass their `ctx` into this function.

Reply:
If I see it correctly this is also the case for the other added functions, please change them as well use the provided context.